storage/block/block.go

```diff
@@ -326,40 +332,45 @@ func (dbb *databaseSeriesBlocks) AddSeries(other DatabaseSeriesBlocks) {
 
 // MinTime returns the min time of the blocks contained.
 func (dbb *databaseSeriesBlocks) MinTime() time.Time {
-	return dbb.min
+	return time.Unix(0, dbb.min)
 }
 
 // MaxTime returns the max time of the blocks contained.
 func (dbb *databaseSeriesBlocks) MaxTime() time.Time {
-	return dbb.max
+	return time.Unix(0, dbb.max)
 }
 
 func (dbb *databaseSeriesBlocks) BlockAt(t time.Time) (DatabaseBlock, bool) {
-	b, ok := dbb.elems[t]
+	b, ok := dbb.elems[t.UnixNano()]
 	return b, ok
 }
 
 func (dbb *databaseSeriesBlocks) AllBlocks() map[time.Time]DatabaseBlock {
-	return dbb.elems
+	res := make(map[time.Time]DatabaseBlock, len(dbb.elems))
```

Comment:
Yeah we can't create a new map each time, this is called on some huge maps. A copy each time is extremely expensive (think millions of IDs all getting `AllBlocks(...)` called on it.